packages/core/src/App/Containers/Layout/header/header.tsx

```diff
@@ -2,26 +2,39 @@ import React from 'react';
 import { useLocation } from 'react-router-dom';
 import { routes } from '@deriv/shared';
 import { observer, useStore } from '@deriv/stores';
+import { useFeatureFlags, useStoreWalletAccountsList } from '@deriv/hooks';
 import DefaultHeader from './default-header.jsx';
 import DTraderHeader from './dtrader-header.jsx';
 import TradersHubHeader from './traders-hub-header';
+import DTraderHeaderWallets from './dtrader-header-wallets';
+import { useReadLocalStorage } from 'usehooks-ts';
 
 const Header = observer(() => {
     const { client } = useStore();
-    const { is_logged_in } = client;
+    const { is_logged_in, setAccounts, setLoginId } = client;
     const { pathname } = useLocation();
     const traders_hub_routes =
         [routes.traders_hub, routes.account, routes.cashier, routes.wallets, routes.compare_cfds].includes(pathname) ||
         pathname.startsWith(routes.compare_cfds);
 
+    const clientAccounts = useReadLocalStorage('client.accounts');
+    const { is_next_wallet_enabled } = useFeatureFlags();
+    const { has_wallet } = useStoreWalletAccountsList();
+    const should_show_wallets = is_next_wallet_enabled && has_wallet;
+
+    React.useEffect(() => {
```

Comment:
whats the responsibility of Layout/header compononent? is it a valid place to handle account synchronisation in header layout header component?

Reply:
@wojciech-deriv good question, but I didn't find better place for this. I can't add useEffect to the store and by this cause it must be in the react component which mounted when the user in trader's hub. I'm open to your suggestions :)

Reply:
cc @markwylde-deriv

Reply:
what about putting it to store? this hook effectively is triggered by store (or, by changes from the store) @markwylde-deriv @sergei-deriv ?